pages/admin.js

```diff
@@ -0,0 +1,248 @@
+import Layout from '../components/Layout'
+import Head from 'next/head'
+import useSWR from 'swr'
+
+import { useEffect, useState } from 'react'
+import { useUser } from '../context/userContext'
+import cookie from 'js-cookie';
+
+const fetcher = (url) => fetch(url).then((res) => res.json())
+
+const token = cookie.get("firebaseToken")
+
+export default function Hours() {
+
+    var dayToHours;
+    var dayObjects = [];
+
+    /**
+     * Fetch the current hours of the pantry and convert them to a map with
+     * the format being day:hours
+     */
+    const fetchHours = async () => {
+
+        await fetch('/api/admin/getHours', {
+            method: 'GET',
+            headers: { 'Content-Type': "application/json" }
+        })
+            .then(function (res) {
+                res.json().then(json => {
+                    dayToHours = json.message
+                    dayToHours = new Map(dayToHours)
+                    createDayObjects()
+                });
+
+            })
+            .catch(function (error) {
+            });
+
+
+
+    }
+
+    /**
+     * Convert the map into objects with day and hour properties
+     * 
+     */
+    const createDayObjects = () => {
+        const days = ["sunday", "monday", "tuesday", "wednesday", "thursday", "friday", "saturday"]
+
+        for (let day in days) {
+            const dayObject = new Object();
+            dayObject.day = days[day];
+            dayObject.hours = dayToHours.get(days[day])
+            dayObjects.push(dayObject);
+        }
+
+    }
+
+    useEffect(() => {
+
+        fetchHours();
+    }, []);
+
+
+    const [inEditMode, setInEditMode] = useState({
+        status: false,
+        rowKey: null
+    });
+
+    const [hours, setHours] = useState(null);
+
+    /**
+     *
+     * @param day - The day we are trying to edit
+     * @param currentHours - The current hours
+     */
+    const onEdit = ({ day, currentHours }) => {
+        setInEditMode({
+            status: true,
+            rowKey: day
+        })
+        setHours(currentHours)
+
+    }
+
+
+
+    /**
+     *
+     * @param day -The day we are trying to change
+     * @param newHours- The new hours of the day
+     */
+    const onSave = ({ day, newHours }) => {
+
+        fetch('/api/admin/updateHours', {
+            method: 'POST',
+            body: JSON.stringify({
+                "day": day,
+                "hours": newHours,
+            }),
+            headers: { 'Content-Type': "application/json", 'Authorization': token }
+        })
+            .then(response => response.json())
+            .then(json => {
+                // reset inEditMode and unit price state values
+                onCancel();
+
+                // fetch the updated data
+                fetchHours();
```

Comment:
I think there might be a way to have `fetchHours` update the `hours` state + some render function in `useEffect` and not have to call `fetchHours` again here. Doesn't really matter though.